concepts/modularization/migration.md

```diff
@@ -0,0 +1,36 @@
+# Migration plan
+
+Migration plan will be executed several times until all modules are moved from reconciler to lifecycle-manager. Is it possible to migrate several modules at once but it is not recommended. The migration plan uses `serverless` module as an example.
+
+## Prerequisite: enable lifecycle-manager
+- KEB creates Kyma resources in the control plane. Lifecycle-manager is deployed in KCP and enabled for all clusters with empty module list.
+
+## Enable module for new clusters (first)
+
+First step is to enable module installation with lifecycle manager for new clusters. Existing clusters are still managed by reconciler.
+- Serverless operator(manager) is ready and module template is available in the fast/regular channel. 
+- New Kyma version is enabled in KEB (without serverless module). New clusters do not have serverless until customer enabled it in Kyma resource.
+
+## Update existing clusters
+
+Migration can be stopped at any moment if any problem occurs. Not migrated clusters will still use serverless reconciler (old Kyma version with serverless)
+
+1. Upgrade the clusters to the new Kyma version (without serverless)
+
+2. Create CRD and default resource for serverless manager in the SKR (we can have different CRs for different plans/profiles if needed)
+
+
+![migration](assets/migration.drawio.svg)
+
+# Validation scenarios
+
+The migration process should be resilient and validated against possible user actions that could cause problems. It is mainly about turning on and off the module that is being migrated:
```

Comment:
I believe it would be safest to apply the Module in KCP during the migration period with `noModuleCopy` set to `false`. This will cause the module to be reflected in the SKR and even if it is removed by the user, the module will still be active and avoid side effects

Reply:
I believe `noModuleCopy` set to `false` means `moduleCopy=true` ;-) (I suggest to avoid negative boolean variable names)
But if we set `noModuleCopy=true` it can work. When users try to enable module on their own it will have no effect. Module CR is not created in the SKR so the module operator is not active yet (assuming the module operator is not doing clean-up if it cannot find any module CR in the cluster)

Reply:
reason for negative is so that the default bool value can be used and we don't have to write a defaulting web hook to set it to true every time.

I agree with your sentiment on the Module CR. I'll talk with @janmedrek to see if we should prioritise the feature earlier as we originally thought to allow this or if we could go with an alternative.